share/spice/currency/currency.js

```diff
@@ -29,29 +222,150 @@
         "xbt": 8
     };
     
-    // Resize the size of the outer container if the content of the inner container
-    // overflows.
-    function resize() {        
-        var resultHeight = $(".zci--currency-result").outerHeight();
-        
-        if(resultHeight > 65) {
-            $(".zci--currency-container").css("height", "9em");
-        } else {
-            $(".zci--currency-container").css("height", "5em");
-        }
-    }
-    
-    // Change the look of the mobile view if the content overflows.
-    function resizeMobile() {
-        var tileHeight = $(".zci--currency .tile--s").outerHeight();
 
-        if(tileHeight > 155) {
-            $(".zci--currency .tile--s").addClass("large").removeClass("small");
-        } else {
-            $(".zci--currency .tile--s").addClass("small").removeClass("large");
-        }
-    }
-    
+    // Currency Converter
+    var Converter = {
+
+        from_currency: undefined,
+        to_currency: undefined,
+        rate: undefined,
+        inverseRate: undefined,
+
+        //
+        // Retrieves conversion info from JSONP payload
+        //
+
+        getConversionRate: function(payload) {
+            return +$(payload["conversion-rate"].split(" ")).get(-2);
+        },
+
+        getInverseConversionRate: function(payload) {
+            return +$(payload["conversion-inverse"].split(" ")).get(-2);
+        },
+
+        getFromCurrencyName: function(payload) {
+            return payload["from-currency-symbol"];
+        },
+
+        getToCurrencyName: function(payload) {
+            return payload["to-currency-symbol"];
+        },
+
+        getConversionLabel: function(payload) {
+            return payload["conversion-rate"];
+        },
+
+        getInverseConversionLabel: function(payload) {
+            return payload["conversion-inverse"]
+        },
+
+        //
+        // Calculates the rates
+        //
+
+        calculateRate: function() {
+            if($currency_input_left.val() !== '' && /\d+/.test($currency_input_left.val())) {
```

Comment:
I have now allowed `,` in the input and break with alphabetical characters like I do conversions IA. It certainly is a can of worms. I have people demanding we support this in the calculator, but to do so we need to support mixed types. I think we need a bigger solution when it comes to treating `,` as `.`